test/sql/test_deprecations.py

```diff
@@ -135,6 +153,105 @@ def test_table_useexisting(self):
                 autoload_with=testing.db,
             )
 
+    def test_case_sensitive(self):
+        reg = functions._registry
+        cs_reg = functions._case_sensitive_reg
+
+        class MYFUNC(GenericFunction):
+            type = DateTime
+
+        assert isinstance(func.MYFUNC().type, DateTime)
+        assert isinstance(func.MyFunc().type, DateTime)
+        assert isinstance(func.mYfUnC().type, DateTime)
+        assert isinstance(func.myfunc().type, DateTime)
+
+        assert "myfunc" in reg['_default']
```

Comment:
**mike bayer** (zzzeek) wrote:

nit: for all the "in" / "not in" we have little functions in_(), not_in_() which can be imported from "sqlalchemy.testing".   We should probably have isinstance / not_isinstance too.    There's a not-well-attended to convention that we don't want to use "assert " in test suites even though yes it is all over the place.

Reply:
Ok, I will update that.